### ISSUE TITLE
Helm upgrade failed from 4.0 -> 4.1 with old CRDs

### ISSUE THREAD

**Issue by cmccab-icims** [SumoLogic/sumologic-kubernetes-collection#3397]
Trying to upgrade from 4.01 to 4.1.0 latest version with helm chart gives error below in Command & Error section. I know Helm upgrade doesn't upgrade CRD's but trying to understand what I need to do to fix this issue without destroying the release and reinstalling. Sumologic 2.X was originally installed on this cluster but recently upgraded from 1.21 to 1.27. Install of sumo 4.0.1 was successful which is confusing as to why this upgrade would be an issue?

Command & Error:
`
$ helm upgrade sumologic sumologic/sumologic -f prd-values.yaml
Error: UPGRADE FAILED: unable to build kubernetes objects from current release manifest: [resource mapping not found for name: "sumologic-sumologic-otelcol-instrumentation" namespace: "sumologic" from "": no matches for kind "HorizontalPodAutoscaler" in version "autoscaling/v2beta2"
ensure CRDs are installed first, resource mapping not found for name: "sumologic-sumologic-traces-gateway" namespace: "sumologic" from "": no matches for kind "HorizontalPodAutoscaler" in version "autoscaling/v2beta2"
ensure CRDs are installed first, resource mapping not found for name: "sumologic-sumologic-otelcol-logs" namespace: "sumologic" from "": no matches for kind "HorizontalPodAutoscaler" in version "autoscaling/v2beta2"
ensure CRDs are installed first]
`

EKS Cluster details:
`
$ k version --short
Client Version: v1.27.7
Kustomize Version: v5.0.1
Server Version: v1.27.7-eks-4f4795d
`

Helm details:
`
$ helm list
NAME     	NAMESPACE	REVISION	UPDATED                               	STATUS  	CHART          	APP VERSION
sumologic	sumologic	1       	2023-10-28 03:07:45.640441 -0400 -0400	deployed	sumologic-4.0.1	4.0.1
`

CRD details:
`
k get crd
NAME                                         CREATED AT
alertmanagerconfigs.monitoring.coreos.com    2021-10-08T06:02:01Z
alertmanagers.monitoring.coreos.com          2021-10-08T06:02:02Z
cninodes.vpcresources.k8s.aws                2023-08-13T16:31:22Z
eniconfigs.crd.k8s.amazonaws.com             2019-01-08T21:36:30Z
instrumentations.opentelemetry.io            2023-10-28T06:59:46Z
opentelemetrycollectors.opentelemetry.io     2023-10-28T06:59:45Z
podmonitors.monitoring.coreos.com            2021-10-08T06:02:03Z
policyendpoints.networking.k8s.aws           2023-11-09T02:32:52Z
probes.monitoring.coreos.com                 2021-10-08T06:02:04Z
prometheuses.monitoring.coreos.com           2021-10-08T06:02:06Z
prometheusrules.monitoring.coreos.com        2021-10-08T06:02:06Z
securitygrouppolicies.vpcresources.k8s.aws   2020-09-16T02:38:03Z
servicemonitors.monitoring.coreos.com        2021-10-08T06:02:07Z
thanosrulers.monitoring.coreos.com           2021-10-08T06:02:08Z
`

**Comment by astencel-sumo** [comment]
Thanks for reporting this @cmccab-icims. I believe the issue you've encountered is a result of Helm limitation, see this issue in Helm project for more details:

- https://github.com/helm/helm/issues/7219

The easiest way to resolve this is to uninstall and reinstall the chart. Is this a path that you can follow?

**Comment by cmccab-icims** [comment]
Thanks for the quick response @astencel-sumo. I definitely could run through an uninstall and reinstall. Deleting the release wasn't a problem but deleting the namespace I ran into issues in another cluster. I needed to delete old resources which I've documented below for anyone who is running into a similar problem. I do not know if it was wholly necessary but inorder to delete the existing namespace alongside the release I had to delete the following:
Namespace = 'sumologic'
`
kubectl delete ValidatingWebhookConfiguration sumologic-opentelemetry-operator-validation

kubectl delete MutatingWebhookConfiguration sumologic-opentelemetry-operator-mutation

kubectl delete clusterrolebinding sumologic-fluent-bit sumologic-metrics-server sumologic-metrics-server-auth-delegator sumologic-metrics-server-sumologic sumologic-metrics-server-sumologic-auth-delegator sumologic-opentelemetry-operator-manager sumologic-opentelemetry-operator-proxy sumologic-sumologic

kubectl delete clusterrole sumologic-metrics-server-sumologic sumologic-metrics-server-sumologic-view sumologic-opentelemetry-operator-manager sumologic-opentelemetry-operator-metrics sumologic-opentelemetry-operator-proxy sumologic-sumologic

kubectl delete PriorityClass sumologic-sumologic-priorityclass

kubectl delete apiservices.apiregistration.k8s.io v1beta1.metrics.k8s.io

kubectl delete secret sumologic

kubectl delete ns sumologic
`
